zebra/interface.c

```diff
@@ -92,6 +90,23 @@ static int if_zebra_speed_update(struct thread *thread)
 	}
 
 	if (changed || new_speed == UINT32_MAX) {
+#define SPEED_UPDATE_COUNT_MAX 80
+		/*
+		 * Some interfaces never actually have an associated speed
+		 * with them ( I am looking at you bridges ).
+		 * So instead of iterating forever, let's give the
+		 * system 4 minutes to try to figure out the speed
+		 * if after that it it's probably never going to become
+		 * useful.
+		 * Since I don't know all the wonderful types of interfaces
+		 * that may come into existence in the future I am going
+		 * to not update the system to keep track of that.  This
+		 * is far simpler to just stop trying after 4 minutes
+		 */
+		if (new_speed == UINT32_MAX && zif->speed_update_count == 40)
```

Comment:
How does 40 translates to 4 minutes ? Does this function get called for every 6 seconds ?

Reply:
because I can't do math :( .. each timer is 5 seconds and we want 4 minutes of that so I am changing the value to 46

Reply:
Thanks for fixing it.